R/crew_launcher_lsf.R

```diff
@@ -0,0 +1,287 @@
+#' @title `r lifecycle::badge('experimental')` Create a launcher with
+#'   LSF workers.
+#' @export
+#' @family launchers
+#' @description Create an `R6` object to launch and maintain
+#'   workers as LSF jobs.
+#' @details WARNING: the `crew.cluster` LSF plugin is experimental
+#'   and has not actually been tested on a LSF cluster. Please proceed
+#'   with caution and report bugs to
+#'   <https://github.com/wlandau/crew.cluster>.
+#'
+#'   To launch a LSF worker, this launcher
+#'   creates a temporary job script with a call to `crew::crew_worker()`
+#'   and submits it as an LSF job with `sbatch`. To see most of the lines
+#'   of the job script in advance, use the `script()` method of the launcher.
+#'   It has all the lines except for the job name and the
+#'   call to `crew::crew_worker()`, both of
+#'   which will be inserted at the last minute when it is time
+#'   to actually launch a worker.
+#' @inheritParams crew_launcher_cluster
+#' @param lsf_log_output Character of length 1, file pattern to control
+#'   the locations of the LSF worker log files. By default, both standard
+#'   output and standard error go to the same file.
+#'   `lsf_log_output = "crew_log_%J.log"` translates to a line of
+#'   `#BSUB-o crew_log_%J.log` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_output = NULL` to omit this line from the job script.
+#' @param lsf_log_error Character of length 1, file pattern for standard
+#'   error. `lsf_log_error = "crew_error_%J.err"` translates to a line of
+#'   `#BSUB-e crew_error_%J.err` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_error = NULL` to omit this line from the job script.
+#' @param lsf_memory_megabytes_per_cpu Positive numeric of length 1
+#'   with the megabytes of memory required.
+#'   `lsf_memory_megabytes_per_cpu = 4096`
+#'   translates to a line of `#BSUB-M 4096`
+#'   in the LSF job script.
+#'   `lsf_memory_megabytes_per_cpu = NULL` omits this line.
+#' @param lsf_cpus_per_task Optional positive integer of length 1,
+#'   number of CPUs for the worker.
+#'   `lsf_cpus_per_task = 4` translates
+#'   to a line of `#BSUB-n 4` in the LSF job script.
```

Comment:
From https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=o-n-1, it looks like `-n` controls the number of tasks rather than the number of CPUs or CPU cores. Is there a way to set a number of cores to make available for the job? Or is that not how LSF works?

In SGE the user can request a given amount of memory and cores and then they get exclusive access to those resources for the job. I recall hearing that LSF does more sharing, e.g. two users can share the same CPU cores depending on the load at the moment.

Reply:
On the LSF systems that I've used, `-n` has been used to specify the number of cores. This is how the default `clustermq` LSF template works, too: https://github.com/mschubert/clustermq/blob/master/inst/LSF.tmpl

Reply:
In that case, what you have looks great! Can we call it `lsf_cores`?